keps/sig-api-machinery/2161-apiserver-default-labels/README.md

```diff
@@ -0,0 +1,452 @@
+
+No
+
+### Troubleshooting
+
+The Troubleshooting section currently serves the `Playbook` role. We may consider
+splitting it into a dedicated `Playbook` document (potentially with some monitoring
+details). For now, we leave it here.
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How does this feature react if the API server and/or etcd is unavailable?**
+
+* **What are other known failure modes?**
+  For each of them, fill in the following information by copying the below template:
+  - [Failure mode brief description]
+    - Detection: How can it be detected via metrics? Stated another way:
+      how can an operator troubleshoot without logging into a master or worker node?
+    - Mitigations: What can be done to stop the bleeding, especially for already
+      running user workloads?
+    - Diagnostics: What are the useful log messages and their required logging
+      levels that could help debug the issue?
+      Not required until feature graduated to beta.
+    - Testing: Are there any tests for failure mode? If not, describe why.
+
+* **What steps should be taken if SLOs are not being met to determine the problem?**
+
+[supported limits]: https://git.k8s.io/community//sig-scalability/configs-and-limits/thresholds.md
+[existing SLIs/SLOs]: https://git.k8s.io/community/sig-scalability/slos/slos.md#kubernetes-slisslos
+
+## Implementation History
+* 2020-11-22 - Creation of the issue and the KEP 
+* 2020-11-30 - https://github.com/kubernetes/kubernetes/pull/96968/files example PR implementing this
+
+## Drawbacks
+
+* "It's kind of a special hack. The fact that it isn't generalized or generalizable 
+makes it feel special." - @thockin
+
+## Alternatives
+
+### Add new select-by-name capabilities to policy APIs
+
+#### API Machinery modifications and workarounds
+
+* Add an 'in memory' virtual label that is defaulted by the apiserver but not stored in etcd, this has the drawback of diverging the apiserver from data in an etcd backup, making inspection and recovery of offline clusters more complicated.
+* Add a new syntax for selectors on labels that allow them to dynamically select things in fields,  This has the drawback of being prohibitively complicated from a perspective of community convergence on what the boundaries of such a syntax might actually be.
+
+#### API modifications at the NetworkPolicy level
+
+* The original proposal was to create a new kind of selector for NetworkPolicies, this of course has many drawbacks, including most of all, the
+need for new API semantics in the networkpolicy specification, which is already tricky to understand for newcomers.
+
+For example, you may need a way to declare namespaceAsName selectors, which is distinct from namespace label selectors.
+Introducing a new field has its own drawbacks: The obvious API, when combined with a podSelector, decreases security
+of a pod - an old client would be more open than users would expect them to be.
```

Comment:
That depends on how you write the persisting logic.  One simple way to avoid is to say that when using this new field, a label selector is set by default which will never match anything.

Reply:
yeah we discussed a few clever workarounds, they were though pretty odd.  ill add this comment as well , WRT workaround possibilities